tests/net/p2p/p2p.cpp

```diff
@@ -358,135 +371,137 @@ namespace TP2P {
       p2pNode8.startDiscovery();
       p2pNode9.startDiscovery();
       p2pNode10.startDiscovery();
-      
-			// After a while, the discovery thread should have found all the nodes and connected between each other.
-      while(p2pDiscoveryNode.getSessionsIDs().size() != 10 ||
-            p2pNode1.getSessionsIDs().size() != 10 ||
-            p2pNode2.getSessionsIDs().size() != 10 ||
-            p2pNode3.getSessionsIDs().size() != 10 ||
-            p2pNode4.getSessionsIDs().size() != 10 ||
-            p2pNode5.getSessionsIDs().size() != 10 ||
-            p2pNode6.getSessionsIDs().size() != 10 ||
-            p2pNode7.getSessionsIDs().size() != 10 ||
-            p2pNode8.getSessionsIDs().size() != 10 ||
-            p2pNode9.getSessionsIDs().size() != 10 ||
-            p2pNode10.getSessionsIDs().size() != 10) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
-      }
-      
 
+      // After a while, the discovery thread should have found all the nodes and connected between each other.
+      auto futureWaitAllNodesConnected = std::async(std::launch::async, [&]() {
+        while(p2pDiscoveryNode.getSessionsIDs().size() != 10 ||
+              p2pNode1.getSessionsIDs().size() != 10 ||
+              p2pNode2.getSessionsIDs().size() != 10 ||
+              p2pNode3.getSessionsIDs().size() != 10 ||
+              p2pNode4.getSessionsIDs().size() != 10 ||
+              p2pNode5.getSessionsIDs().size() != 10 ||
+              p2pNode6.getSessionsIDs().size() != 10 ||
+              p2pNode7.getSessionsIDs().size() != 10 ||
+              p2pNode8.getSessionsIDs().size() != 10 ||
+              p2pNode9.getSessionsIDs().size() != 10 ||
+              p2pNode10.getSessionsIDs().size() != 10) {
+          std::this_thread::sleep_for(std::chrono::milliseconds(5));
+        }
+      });
+
+      REQUIRE(futureWaitAllNodesConnected.wait_for(std::chrono::seconds(5)) != std::future_status::timeout);
+
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
 
-			std::this_thread::sleep_for(std::chrono::milliseconds(100));
-      
       auto nodeDiscoverySessionsIDs = p2pDiscoveryNode.getSessionsIDs();
-			auto node1SessionsIDs = p2pNode1.getSessionsIDs();
-			auto node2SessionsIDs = p2pNode2.getSessionsIDs();
-			auto node3SessionsIDs = p2pNode3.getSessionsIDs();
-			auto node4SessionsIDs = p2pNode4.getSessionsIDs();
-			auto node5SessionsIDs = p2pNode5.getSessionsIDs();
-			auto node6SessionsIDs = p2pNode6.getSessionsIDs();
-			auto node7SessionsIDs = p2pNode7.getSessionsIDs();
-			auto node8SessionsIDs = p2pNode8.getSessionsIDs();
-			auto node9SessionsIDs = p2pNode9.getSessionsIDs();
-			auto node10SessionsIDs = p2pNode10.getSessionsIDs();
-      
+      auto node1SessionsIDs = p2pNode1.getSessionsIDs();
```

Comment:
Why auto instead of `std::vector<NodeID>`?

Reply:
Doesn't really matter, the function naming already tells its type.